### PR TITLE
InfoMax3DModular

### PR DESCRIPTION
**Pull request by tonydavis629**
## Description

The full implementation of 3DInfomax as a ModularTorchModel. Maximizes the mutual information between a 2D GNN and a 3D GNN, then uses the 2D GNN for inference to leverage learned 3D training data with 2D inputs. 

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
 - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
 - [x] Run `mypy -p deepchem` and check no errors
 - [x] Run `flake8 <modified file> --count` and check no errors
 - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by rbharath — COMMENTED** [review]
Did a first pass and generally things look good. We need to rebase this PR. Also we are at 20 passing tests so we should double check CI is clean

File: deepchem/models/tests/test_gnn3d.py [hunk 1]
```diff
@@ -69,3 +77,20 @@ def test_net3d():
...
+
+    model = InfoMax3DModular(hidden_dim=64,
+                             target_dim=10,
+                             aggregators=['sum', 'mean', 'max'],
+                             readout_aggregators=['sum', 'mean'],
+                             scalers=['identity'])
+
+    loss1 = model.fit(data, nb_epoch=1)
+    loss2 = model.fit(data, nb_epoch=9)
+    assert loss1 > loss2
```
Comment (rbharath) [inline deepchem/models/tests/test_gnn3d.py:96]: Do we have save/reload tests for this model? 

The loss reduction here handles basic functionality, but should we add an overfit test?

**Review by rbharath — COMMENTED** [review]
The code looks mostly fine, but have a question about tests

**Review by rbharath — APPROVED** [review]
LGTM.

@tonydavis629 Feel free to merge in once you rebase/remove merge conflicts, and verify that the CI is good
